### PR TITLE
What's new OpenAPI 2

### LINKED ISSUE #33016: Completion Enhancements to Microsoft.AspNetCore.OpenAPI
### Description

[Completion Enhancements to Microsoft.AspNetCore.OpenAPI](https://github.com/dotnet/AspNetCore.Docs/issues/32960#issuecomment-2212549719)

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?branch=main&branchFallbackFrom=pr-en-us-32787&view=aspnetcore-8.0#openapi

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/release-notes/aspnetcore-9.0.md

### Document ID

4e75ad25-2c3f-b28e-6a91-ac79a9c683b6

### Article author

@Rick-Anderson

---
[Associated WorkItem - 280591](https://dev.azure.com/msft-skilling/Content/_workitems/edit/280591)

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #33016

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/release-notes/aspnetcore-9.0.md](https://github.com/dotnet/AspNetCore.Docs/blob/be0aa47b7cbb84d680485aa41ccce16b5885c129/aspnetcore/release-notes/aspnetcore-9.0.md) | [What's new in ASP.NET Core 9.0](https://review.learn.microsoft.com/en-us/aspnet/core/release-notes/aspnetcore-9.0?branch=pr-en-us-33017) |

### CODE REVIEW

**Comment by Rick-Anderson** [comment]
@tdykstra  apply your suggestions, then S&M. If I disagree with any, I'll do another PR.

File: aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md [hunk 1]
```diff
@@ -1,3 +1,128 @@
...
\ No newline at end of file
+ASP.NET Core's OpenAPI support is now more accessible and user-friendly. The OpenAPI APIs are shipped as an independent package, separate from the shared framework. This meant that developers didn't have the convenience of code-completion features like Intellisense for OpenAPI APIs.
+
+Recognizing this gap, the ASP.NET Core team has introduced a new completion provider and codefixer in .NET 9 Preview 6. These tools are designed to facilitate the discovery and use of OpenAPI APIs, making it easier for developers to integrate OpenAPI into their projects. The completion provider offers real-time code suggestions, while the codefixer assists in correcting common mistakes and improving API usage. This enhancement is part of our ongoing commitment to improve the developer experience and streamline API-related workflows.
```
Comment (tdykstra) [inline aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md:5]:
```suggestion
Recognizing this gap, we have introduced a new completion provider and code fixer. These tools are designed to facilitate the discovery and use of OpenAPI APIs, making it easier for developers to integrate OpenAPI into their projects. The completion provider offers real-time code suggestions, while the code fixer assists in correcting common mistakes and improving API usage. This enhancement is part of our ongoing commitment to improve the developer experience and streamline API-related workflows.
```

File: aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md [hunk 2]
```diff
@@ -1,3 +1,128 @@
-### Completion enhancements and package install recommendations for OpenAPI package
+### Intellisense completion enhancements for OpenAPI package
 
-ASP.NET Core's OpenAPI support ships in an independent package outside of the shared framework. This means that it's difficult for users to discover built-in OpenAPI API's with code-completion aids like Intellisense. In .NET 9 Preview 6, we've shipped a completion provider/codefixer combination to help users discover built-in OpenAPI support.
\ No newline at end of file
+ASP.NET Core's OpenAPI support is now more accessible and user-friendly. The OpenAPI APIs are shipped as an independent package, separate from the shared framework. This meant that developers didn't have the convenience of code-completion features like Intellisense for OpenAPI APIs.
```
Comment (tdykstra) [inline aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md:3]:
```suggestion
ASP.NET Core's OpenAPI support is now more accessible and user-friendly. The OpenAPI APIs are shipped as an independent package, separate from the shared framework. Until now, this meant that developers didn't have the convenience of code-completion features like Intellisense for OpenAPI APIs.
```

File: aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md [hunk 3]
```diff
@@ -1,3 +1,128 @@
...
+	  },
+	  "createdOn": {
+	    "type": "string",
+	    "format": "date-time"
+	  }
+	}
+}
+```
+
+### Support for schema transformers on OpenAPI document
```
Comment (tdykstra) [inline aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md:79]:
```suggestion
### Support for schema transformers on OpenAPI documents
```

File: aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md [hunk 4]
```diff
@@ -1,3 +1,128 @@
...
+
+In .NET 9 Preview 6, built-in OpenAPI support ships with support for schema transformers that can be used to modify schemas generated by [System.Text.Json](/dotnet/api/system.text.json) and the OpenAPI implementation. Like document and operation transformers, schema transformers can be registered on the [OpenApiOptions](https://source.dot.net/#Microsoft.AspNetCore.OpenApi/Services/OpenApiOptions.cs,c0a8b420f4ce6918) object. For example, the following code sample demonstrates using a schema transformer to add an example to type's schema.
```
Comment (tdykstra) [inline aspnetcore/release-notes/aspnetcore-9/includes/openAPI_completion.md:81]:
```suggestion
Built-in OpenAPI support now ships with support for schema transformers that can be used to modify schemas generated by [System.Text.Json](/dotnet/api/system.text.json) and the OpenAPI implementation. Like document and operation transformers, schema transformers can be registered on the [OpenApiOptions](https://source.dot.net/#Microsoft.AspNetCore.OpenApi/Services/OpenApiOptions.cs,c0a8b420f4ce6918) object. For example, the following code sample demonstrates using a schema transformer to add an example to a type's schema.
```

**Review by tdykstra — APPROVED** [review]

**Pull request merged**